File: tools/oracle_diff/fixtures/typescript/basic/src/shapes.ts
export class Circle {
    r: number;
    constructor(r: number) { this.r = r; }
    area(): number { return 3 * this.r * this.r; }
    perimeter(): number { return 6 * this.r; }
    describe(): number { return this.area() + this.perimeter(); }
}

export class Square {
    s: number;
    constructor(s: number) { this.s = s; }
    area(): number { return this.s * this.s; }
    perimeter(): number { return 4 * this.s; }
    describe(): number { return this.area() + this.perimeter(); }
}

File: tools/oracle_diff/fixtures/python/basic/src/shapes.py
class Circle:
    def __init__(self, r):
        self.r = r

    def area(self):
        return 3 * self.r * self.r

    def describe(self):
        return self.area()  # self.method() → Circle.area, not Square.area


class Square:
    def __init__(self, s):
        self.s = s

    def area(self):
        return self.s * self.s

    def describe(self):
        return self.area()  # self.method() → Square.area
